src/ConnectedMode/Binding/BoundConnectionInfoProvider.cs

```diff
@@ -28,24 +28,22 @@
 
 namespace SonarLint.VisualStudio.ConnectedMode.Binding
 {
-    [Export(typeof(IBindingInfoProvider))]
+    [Export(typeof(IBoundConnectionInfoProvider))]
     [PartCreationPolicy(CreationPolicy.Shared)]
-    internal class BoundConnectionInfoProvider : IBindingInfoProvider
+    internal class BoundConnectionInfoProvider : IBoundConnectionInfoProvider
     {
-        private readonly IUnintrusiveBindingPathProvider unintrusiveBindingPathProvider;
-        private readonly ISolutionBindingFileLoader solutionBindingFileLoader;
+        private readonly ISolutionBindingRepository solutionBindingRepository;
         private readonly IThreadHandling threadHandling;
 
         [ImportingConstructor]
-        public BoundConnectionInfoProvider(IUnintrusiveBindingPathProvider unintrusiveBindingPathProvider, ISolutionBindingFileLoader solutionBindingFileLoader)
-            : this(unintrusiveBindingPathProvider, solutionBindingFileLoader, ThreadHandling.Instance)
+        public BoundConnectionInfoProvider(ISolutionBindingRepository solutionBindingRepository)
+            : this(solutionBindingRepository, ThreadHandling.Instance)
         {
         }
 
-        internal BoundConnectionInfoProvider(IUnintrusiveBindingPathProvider unintrusiveBindingPathProvider, ISolutionBindingFileLoader solutionBindingFileLoader, IThreadHandling threadHandling)
+        internal BoundConnectionInfoProvider(ISolutionBindingRepository solutionBindingRepository, IThreadHandling threadHandling)
```

Comment:
This class could have been a method of the repo

Reply:
Repo is about read/write operations and more generic, but this class is about getting the unique connections for sloop. I would argue they are logically different things.

Reply:
Still, this could be an extension method even if you don't want to make it a part of the main interface. This piece of functionality is too small to warrant a separate class

Reply:
I don't see having a small class as a problem. This class tackles a separate problem, I might merge the functionality here with converting to SLOOP Dto's (not sure yet how it'll be)

Reply:
Merging with dtos makes more sense